wagtail_transfer/streamfield.py

```diff
@@ -0,0 +1,118 @@
+from wagtail.core.blocks import RichTextBlock, ChooserBlock
+
+from .richtext import get_reference_handler
+from functools import partial
+
+
+def map_over_json(block, stream, func):
+    """Apply a function, func, to each of the base blocks' values (ie not Struct, List, Stream) of a StreamField in
+    list of dicts (imported json) format. Block is the block object, stream is its value."""
+    try:
+        # test if behaviour is like a Struct- or StreamBlock, and try to recursively map over child blocks if so
```

Comment:
Rather than 'heuristically' identifying the patterns for StructBlock / StreamBlock / ListBlock, I think we could make this more robust and extensible by giving those block types their own BlockHandlers, and making `map_over_json` part of the BlockHandler API. Happy to save that for a later cleanup/optimisation PR though.